lib/order_handler.rb

```diff
@@ -0,0 +1,60 @@
+class OrderHandler
+
+  ERROR_MESSAGE = "Sorry, your order was invalid."
+
+  def initialize(menu, order_class, messenger = Messenger.new)
+    @menu = menu
+    @order_class = order_class
+    @messenger = messenger
+  end
+
+  def handle_order(order_string, customer_total, time = Time.now)
+    create_new_order(time)
+    add_all_to_order(order_string)
+    fail(ERROR_MESSAGE) if invalid_total?(customer_total)
+    send_order_message
+    order_success
+  end
+
+  private
+
+  def create_new_order(time)
+    @current_order = @order_class.new(time)
+  end
+
+  def add_all_to_order(order_string)
+    order_arr = split_order_string(order_string)
+    order_arr.each_cons(2) do |item_arr| 
+      item_arr[1].to_i.times { @current_order.add(get_dish(item_arr[0])) }
+    end
+  end
+
+  def split_order_string(order_string)
```

Comment:
This method could make the item and the quantity separate variables so that you don't need to split the string on `*`